code/modules/mob/living/carbon/human/life.dm

```diff
@@ -331,13 +338,186 @@
 
 		oxygen_alert = max(oxygen_alert, 1)
 		return 0
-	var/obj/item/organ/internal/lungs/L = internal_organs_by_name[BP_LUNGS]
-	if(L && L.handle_breath(breath))
-		failed_last_breath = 0
+
+	if(get_organ_efficiency(OP_LUNGS))
+		failed_last_breath = !handle_breath_lungs(breath)
 	else
 		failed_last_breath = 1
 	return 1
 
+/mob/living/carbon/human/proc/handle_breath_lungs(datum/gas_mixture/breath)
+	if(!breath)
+		return FALSE
+	//vars - feel free to modulate if you want more effects that are not gained with efficiency 
+	var/breath_type = species.breath_type ? species.breath_type : "oxygen"
+	var/poison_type = species.poison_type ? species.poison_type : "plasma"
+	var/exhale_type = species.exhale_type ? species.exhale_type : 0
+
+	var/min_breath_pressure = species.breath_pressure
+
+	var/safe_exhaled_max = 10
+	var/safe_toxins_max = 0.2
+	var/SA_para_min = 1
+	var/SA_sleep_min = 5
+
+	var/lung_efficiency = get_organ_efficiency(OP_LUNGS)
+
+	var/safe_pressure_min = min_breath_pressure // Minimum safe partial pressure of breathable gas in kPa
+	// Lung damage increases the minimum safe pressure.
+	if(lung_efficiency < 50)
+		safe_pressure_min *= 1.5
+	else if(lung_efficiency < 80)
+		safe_pressure_min *= 1.25
+
+	var/breath_pressure = (breath.total_moles*R_IDEAL_GAS_EQUATION*breath.temperature)/BREATH_VOLUME
+
+	var/failed_inhale = 0
+	var/failed_exhale = 0
+
+	var/inhaling = breath.gas[breath_type]
+	var/poison = breath.gas[poison_type]
+	var/exhaling = exhale_type ? breath.gas[exhale_type] : 0
+
+	var/inhale_pp = (inhaling/breath.total_moles)*breath_pressure
```

Comment:
Instead of the damaged lungs increasing the safe minimum pressure,I would like the efficiency of your lungs to affect the amount of air you breathe., since it is more to my liking and I also think I have read something about this in the documentation. however, this is just a comment.

Reply:
This will happen, but only when blood/oxy/nutriment requirements for the organs are done, there will come more PR-s after this one.